packages/medusa/src/api/routes/admin/inventory-items/utils/join-variants.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  ProductVariantInventoryService,
+  ProductVariantService,
+} from "../../../../../services"
+import { InventoryItemDTO } from "../../../../../types/inventory"
+import { ProductVariant } from "../../../../../models"
+
+export const getVariantsByInventoryItemId = async (
+  inventoryItems: InventoryItemDTO[],
+  productVariantInventoryService: ProductVariantInventoryService,
+  productVariantService: ProductVariantService
+) => {
+  const variantInventory = await productVariantInventoryService.listByItem(
+    inventoryItems.map((item) => item.id)
+  )
+
+  const inventoryVariantMap = variantInventory.reduce((acc, curr) => {
+    // We only allow one variant per inventory item for now
+    // in the future this should be expanded to handle an array of variants
```

Comment:
suggestion: add a todo

Reply:
I simplified a bit these helper functions and added support to multiple variants per inventory item.
I appreciate if I can have a second look at this @srindom @adrien2p @pKorsholm